test/TESTING_WITH_DOCKER.md

```diff
@@ -0,0 +1,131 @@
+# Testing With Docker
+
+These instructions will guide you through the process of setting up Docker
+to perform tests while developing on the New Relic Agent. Doing so can provide
+you with a consistent experience shared with other developers that helps to
+avoid developer computer specific issues.
+
+*NOTE:* Currently, fewer than 100% of the functional tests pass when ran under
+Docker. We are working to address the related compatibility issues.
+
+
+## Install Docker
+
+You will need to have [Docker Desktop](https://www.docker.com/) installed and
+running on your machine.
+
+If you are using on macOS and using [Homebrew](https://brew.sh/), Docker can be
+installed as a cask via:
+
+```shell
+$ brew cask install docker
+```
+
+and then launched via the `/Applications/Docker.app` launcher that is installed.
+
+For alternatives to using macOS with Homebrew, see Docker's
+[Get Started Guide](https://www.docker.com/get-started).
+
+
+## Clone the project
+
+Use git to clone the [newrelic/newrelic-ruby-agent](https://github.com/newrelic/newrelic-ruby-agent)
+project.
+
+The [Dockerfile](../Dockerfile) and [docker-compose.yml](../docker-compose.yml)
+files are located in the root of the project, and this `TESTING_WITH_DOCKER.md`
+document is located within the `test` subdirectory of the project.
+
+
+## Using just the Dockerfile (unit tests only)
+
+The project `Dockerfile` can be used by itself to run the project unit tests.
+Docker Compose and the project `docker-compose.yml` file will be needed to
+run functional tests which involve communicating with data systems such as
+PostgreSQL and Redis.
+
+To only run the unit tests using `Dockerfile` by itself, first change
+directories to the root of the project, then build an image, and
+finally run a container from the image:
+
+```shell
+$ cd /path/to/project/git/clone
+$ docker build -t newrelic/newrelic-ruby-agent-tester .
+$ docker run -it newrelic/newrelic-ruby-agent-tester
+```
+
+* `build -t <TAG>` applies a tag to the image during building
+* `.` indicates "here" and tells Docker that the `Dockerfile` file can be found
+  in the current directory
+* `run -it` tells Docker to run interactively (`i`) and with a pseudo TTY (`t`)
+
```

Comment:
Should information about running the unit tests in different Rails environments be included here too?

Reply:
Absolutely. I will get that on my TODO list of things to address prior to removing the "in progress" label.

Reply:
The docs now point to `test:all`, which now includes env tests.